helix-term/src/keymap/default.rs

```diff
@@ -86,8 +86,9 @@ pub fn default() -> HashMap<Mode, Keymap> {
 
         "%" => select_all,
         "x" => extend_line_below,
-        "X" => extend_to_line_bounds,
-        "A-x" => shrink_to_line_bounds,
+        "X" => extend_line_above,
+        "A-x" => extend_to_line_bounds,
+        "A-X" => shrink_to_line_bounds,
```

Comment:
I used to use a similar configuration when I used Kakoune. Ultimately I don't think that `extend_line_above` is that useful: if you need to extend upward you can go into select mode and use `A-;` (flip_selections) and `k` to extend above.

It's nice to have `X` readily available because you can use it to blindly delete/change a line (since X behaves differently on an empty line and a line with contents).

Reply:
this change came specifically from a few Matrix conversations where people wanted this functionality, so there is definitely a desire for it. And you can still blindly change a line using xc/Xc? I dont see how X is useful, as it only extends what would have been mid-line selections to full lines. I dont see how this reduces functionality and actually aligns the functionality more

Reply:
> I used to use a similar configuration when I used Kakoune. Ultimately I don't think that `extend_line_above` is that useful: if you need to extend upward you can go into select mode and use `A-;` (flip_selections) and `k` to extend above.

It is apparently important enough to have extend_line_below bound to `x`, but your argument can be be applied to extend_line_below as well? just use v and k to extend down. I don't see why "below" is given importance but "above" isn't? 

`x` is a shortening of a common operation extend_line_below, and extend_line_above isn't that far behind in my mind. full line select, `S-v` in vim would be great in helix too, which might put extend_line_above and extend_line_below lower on the list of important operations. To my knowledge the closest helix has to line select mode is `S-x` although `S-x` will break some muscle memory because it needs to be applied at the end of a selection instead of at the start but I realize that is a personal bias from vim.

Reply:
`x` and `X` behave differently on empty lines:

```
aaa

bbb
```

If you have your cursor on the line between the 'aaa' and 'bbb' lines, `x` extends the selection to include 'bbb' while `X` keeps the selection only on the EOL character on L2. So if you are trying to act consistently on one line, you should use `X` rather than `x` to avoid selecting multiple lines.

> I don't see why "below" is given importance but "above" isn't?

I don't really have a preference between the two; it feels a little wasteful to me to have both though. It's worth considering using `extend_line` which extends in the direction of the selection (forward goes down, backwards goes up) which would eliminate a keybinding here.